### PR TITLE
MAINT: Combine legacy sections of _formatArray

### PR DESCRIPTION
**Pull request by eric-wieser**
This pulls together the changes in #10120, #9815, and #10129 into a more readable fix.

Now it's pretty easy to see the ways in which the legacy behavior was strange.

### CODE REVIEW

File: numpy/core/arrayprint.py [hunk 1]
```diff
@@ -669,28 +669,29 @@ def _formatArray(a, format_function, rank, max_line_len, next_line_prefix,
     else:
         s = '['
         sep = separator.rstrip()
-        line_sep = '\n'*max(rank-1, 1)
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: No need, we already know that `rank` is not `0` or `1`, and if it's any other value `< 1` we have bigger problems.

File: numpy/core/arrayprint.py [hunk 2]
```diff
@@ -669,28 +669,29 @@ def _formatArray(a, format_function, rank, max_line_len, next_line_prefix,
...
-                s += next_line_prefix + summary_insert1.strip() + line_sep
+                s += next_line_prefix + summary_insert + sep + line_sep
 
         for i in range(trailing_items, 1, -1):
             if leading_items or i != trailing_items:
                 s += next_line_prefix
             s += _formatArray(a[-i], format_function, rank-1, max_line_len,
                               " " + next_line_prefix, separator, edge_items,
                               summary_insert, legacy)
-            s = s.rstrip() + sep.rstrip() + line_sep
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: We already called `rstrip` above, so no point doing it again

**Comment by ahaldane** [comment]
Looks good, just needs a rebase.

**Review by mhvk — APPROVED** [review]
Nice, quite a bit clearer.

**Comment by eric-wieser** [comment]
Rebased

**Comment by eric-wieser** [comment]
Tests now passing

**Comment by ahaldane** [comment]
All right, merging. Nice cleanup!

**Pull request merged**